### DISCUSSION TITLE
Using DIDComm with HSM

### DISCUSSION THREAD

**Discussion by koptan** [decentralized-identity/veramo#441] [Q&A]
DIDComm plugin implementation is tightly coupled with Private/Public Key and depending on KMC to sign/encrypt the data.

So is there is any plan to make it less coupled because on our project we are planning to use HSM instead of normal DB?

**Comment by mirceanis** [comment]
That is true, the coupling is too tight.
Definitely there are plans to decouple it, and it is on the high priority list, however I can't provide an estimate.

The current implementation of didcomm will be replaced with didcomm v2 which has other crypto requirements and will likely require some breaking changes in the KMS layer.
